src/main/java/com/enfint/dealservice/service/impl/OfferServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package com.enfint.dealservice.service.impl;
+
+import com.enfint.dealservice.dto.ApplicationStatusHistoryDTO;
+import com.enfint.dealservice.dto.LoanOfferDTO;
+import com.enfint.dealservice.entity.Application;
+import com.enfint.dealservice.exception.ApplicationNotFoundException;
+import com.enfint.dealservice.repository.ApplicationRepository;
+import com.enfint.dealservice.service.OfferService;
+import com.enfint.dealservice.utils.ApplicationStatusEnum;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class OfferServiceImpl implements OfferService {
+
+    private final ApplicationRepository applicationRepository;
```

Comment:
In OFFER-service should be classes that connected only to offer, otherwise it brokes single responsibility principle. You can add extra layer of abstraction and do actions for offer there